### ISSUE TITLE
How to use on Debian 12?

### ISSUE THREAD

**Issue by GitStorageOne** [pgmodeler/pgmodeler#1869]
Hello!

When I execute per-compiled appimage, I get error:  
```
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by ./pgmodeler-1.1.1-linux64.appimage)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libgui.so.1)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libcanvas.so.1)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libconnector.so.1)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libcore.so.1)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libparsers.so.1)
./pgmodeler-1.1.1-linux64.appimage: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_pgmode9ffzPY/lib/libcli.so.1)
```

So, not sure what to do..  
Is it bug or Debian 12 is not supported?

**Comment by rkhaotix** [comment]
Hi 

It seems a Glibc version incompatibility as hinted by the error message. pgModeler libs require 3.4.32, which one is installed in your system?

**Comment by GitStorageOne** [comment]
```
libc6-dev/stable,stable-security,now 2.36-9+deb12u4 amd64 [installed,automatic]
libc6/stable,stable-security,now 2.36-9+deb12u4 amd64 [installed]
```

I guess 2.36?

**Comment by GitStorageOne** [comment]
I remember having a similar problem with my own software when I compiled it using the latest OS.
Many users reported something similar, so I used a separate virtual machine with an old OS to compile.
This library is backwards compatible, so after compiling on an older OS my application worked everywhere.

**Comment by rkhaotix** [comment]
AppImage is designed to be backward compatible with older Glibc version. The problem is with pgModeler build itself. I needed to get rid of that backward compatibility so I could use newer Qt versions. Unfortunately, for the problem you're facing, the solution is to use a newer OS (I don't know how old is Debian 12) but to give you an idea, pgModeler is built on a opensuse 15.5.

**Comment by GitStorageOne** [comment]
It doesn't work either.  
It seems, pre-compiled appimage has a very limited OS support, which also is not listed anywhere.

**Comment by GitStorageOne** [comment]
AFAIK, targeting to 3.4.30 would be OK for most distros.

**Comment by rkhaotix** [comment]
Yeah, I know. I still consider pgModeler AppImages experimental. For the next release I will try to build them on a distro a bit more older than the one I use for development.

**Comment by GitStorageOne** [comment]
> Yeah, I know. I still consider pgModeler AppImages experimental. 

I'm getting the same error with `.run` bundle, so it's more general issue.

>For the next release I will try to build them on a distro a bit more older than the one I use for development.
Latest Ubuntu LTS could be safe choice for build server.

**Comment by rkhaotix** [comment]
Thanks for the suggestion. I'll build a snapshot later today and let you know. ;)

**Comment by rkhaotix** [comment]
Unfortunately, I'm about to ditch (or at least suspend) the use of appimages for pgModeler. It causes more trouble than it is effectively portable. Also, the author [seems to have abandoned the project](https://github.com/probonopd/linuxdeployqt) in favor of a new implementation of [AppImage in Go language](https://github.com/probonopd/go-appimage). 

I'm running a patched older version of linuxdeployqt so it can be compiled in Qt 6 and deploy Qt 6 libraries. The original linuxdeployqt doesn't support Qt 6 and it seems that the author is unwilling to add that support anytime soon.

Anyway, I'll see if the new implementation of AppImage can help me somehow. That's it for now.

**Comment by rkhaotix** [comment]
@jokay I made some changes and build new AppImages on a Kubuntu 20.06 with PostgreSQL 12. Could you please test them on your system?

**Comment by GitStorageOne** [comment]
Well, this is really weakness of the Linux architecture itself.  
I don't understand why developers have to use OLD Linux distribution to compile app.  
One solution is to configure build server which will automatically compile source from proper Linux distro (which should be not too recent - with old GLIBC). While development could happen on the more recent distro for personal preference.
